### PR TITLE
Integer overflow 4516 v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4516

Describe changes:
- Avoids some warnings about integers

Obviously not complete yet... But still mergeable ?

Replaces #6537 with
- formatting
- debug validation added
- another warning found by CI fixed

libhtp-pr: 338

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.wrk.flows_evicted | 5311436 | 4693221 | 113.17% |
| .flow.mgr.rows_maxlen | 115 | 539 | 21.34% | 

Pipeline 4678

**Comment by victorjulien** [comment]
I find "style" in `style: explicit casts to double` do not really be appropriate, but right now also failing to come up with an alternative.

File: src/app-layer.c [hunk 1]
```diff
@@ -303,7 +303,7 @@ static int TCPProtoDetect(ThreadVars *tv,
 {
     AppProto *alproto;
     AppProto *alproto_otherdir;
-    int direction = (flags & STREAM_TOSERVER) ? 0 : 1;
+    bool direction = (flags & STREAM_TOSERVER) ? false : true;
```
Comment (jlucovsky) [inline src/app-layer.c:306]: Suggest `bool direction = (flags & STREAM_TO_SERVER)`

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/6580

File: src/app-layer.c (hunk 1 above)
Comment (catenacyber) [reply src/app-layer.c:306]: Thanks

Comment (catenacyber) [reply src/app-layer.c:306]: Missed a negation

**Pull request closed without merging**